usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -70,13 +71,59 @@ def get_recipients(filters={}):
     return results, page_metadata
 
 
+def duns_count_key_function(view_instance, view_method, request, args, kwargs):
+    validated_payload = TinyShield(view_instance.models).block(request.data)
+
+    award_type = "-"
+    if "award_type" in validated_payload:
+        award_type = validated_payload["award_type"]
+
+    keyword = "-"
+    if "keyword" in validated_payload:
+        keyword = validated_payload["keyword"]
+
+    return ".".join(
+        [
+            award_type,
+            keyword,
+        ]
+    )
```

Comment:
If I understand this correctly, this is creating a separate cache key which is separate from the key generated when the API endpoint is directly requested?

Is this is acceptable it might be helpful to comment on why this is necessary

Reply:
Commented:
>  A custom cache key is created for this endpoint to ensure that only 'award_type' and 'keyword' fields are used from the request body. Other fields that may come in when this view is called from the RecipientList endpoint, such as 'page', should be ignored.

Reply:
I'm still a little confused. The justification for the custom key doesn't explain why the function is creating a string of `<field1>.<field2>` (or even `-.-`), it seems like an odd format.